changelogs/fragments/1353-revert-env-fallback.yml

```diff
@@ -0,0 +1,7 @@
+bugfixes:
+- revert breaking change introduced in 5.2.0 when passing credentials through a mix of environment variables and parameters (https://github.com/ansible-collections/amazon.aws/issues/1353).
+deprecated_features:
+- support for passing both profile and security tokens through a mix of environment variables and parameters has been deprecated and support will be removed in release 6.0.0.
+  After release 6.0.0 it will only be possible to pass either a profile or security tokens, both both at the same time.
```

Comment:
```suggestion
  After release 6.0.0 it will only be possible to pass either a profile or security tokens, regardless of mechanism used to pass them.  To explicitly block a parameter coming from an environment variable pass the empty string as a parameter```